common/primitives/src/ds.rs

```diff
@@ -0,0 +1,38 @@
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::{traits::Get, BoundedVec};
+use orml_utilities::OrderedSet;
+use scale_info::TypeInfo;
+
+/// Extending OrderSet and implement struct OrderedSetExt
+#[derive(TypeInfo, Clone, Decode, Encode, PartialEq, Eq, MaxEncodedLen, Default)]
+#[scale_info(skip_type_params(T, S))]
+pub struct OrderedSetExt<
+	T: Ord + Encode + Decode + MaxEncodedLen + Clone + Eq + PartialEq,
+	S: Get<u32>,
+>(pub OrderedSet<T, S>);
+
+impl<T, S> OrderedSetExt<T, S>
+where
+	T: Ord + Encode + Decode + MaxEncodedLen + Clone + Eq + PartialEq + core::fmt::Debug,
+	S: Get<u32>,
+{
+	/// Create a new empty set
+	pub fn new() -> Self {
+		Self(OrderedSet::<T, S>::new())
+	}
+	/// Create a set from a `Vec`.
+	/// `v` will be sorted and dedup first.
+	pub fn from(bv: BoundedVec<T, S>) -> Self {
+		Self(OrderedSet::<T, S>::from(bv))
+	}
+}
+
+impl<T, S> core::fmt::Debug for OrderedSetExt<T, S>
+where
+	T: Ord + Encode + Decode + MaxEncodedLen + Clone + Eq + PartialEq + core::fmt::Debug,
+	S: Get<u32>,
+{
+	fn fmt(&self, _f: &mut sp_std::fmt::Formatter) -> sp_std::fmt::Result {
+		Ok(())
```

Comment:
Question: our data structures implment some traits such as ```#[derive(TypeInfo, Debug, Clone, Decode, Encode, PartialEq, Default, MaxEncodedLen)]```

Now some third party do not really implement these, do we really want all traits to be extended if not we have to do what i did above

Reply:
if you think we would end up using some other aspects of **orml_utilities** later maybe it makes sense to add this dependency but if not, having this ordered set implementation using sorted vec copied in here with added traits would make more sense.
https://github.com/open-web3-stack/open-runtime-module-library/blob/463635e30cd36b033d5856b1e1eac3fe449c4365/utilities/src/ordered_set.rs

Reply:
I am not against it, but then we will be responsible to maintain it too, adding to our tech debt, since we are using orml in many other cases, ideally we might just use whats out there and put the debt on third party

But i am open to suggestions @enddynayn  you have any feedback on if we just port that code in frequency